### ISSUE TITLE
[java] CyclomaticComplexity marker remains after editing method

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#2247]
**Affects PMD Version:** 

6.20.0

**Rule:**

CyclomaticComplexity

**Description:**

Let's say I have a method that has a cyclomatic complexity of 11.  I save the file and PMD automatically runs.  The rule CyclomaticComplexity flags the method for the problem.

I then edit the method to reduce the cyclomatic complexity to 4.  I save the file and PMD automatically runs.  The violation for CyclomaticComplexity remains even though the method clearly has a reduced cyclomatic complexity.

If I change the name of the method, then the violation for CyclomaticComplexity is cleared.  If I revert the name of the method, then the violation for CyclomaticComplexity returns.

I right-clicked on the Java file and clicked on PMD > Clear Violations.  The violation for CyclomaticComplexity disappears.  I right-clicked on the Java file and clicked on PMD > Check Code.  The violation for CyclomaticComplexity reappears.

Other PMD violations are cleared when I make the appropriate code changes.

If I run `pmd.bat` from the command line, then no PMD violations are flagged.

**Code Sample demonstrating the issue:**

Code before the change...

```
   public static @Nullable String extractIdentifier(String text, MutableInteger position)
   {
      String result;
      int start, pos;
      char check;

      start = position.get();
      pos   = start;

      if (pos >= text.length())
         return(null);

      check = text.charAt(pos);

      if (!Character.isLetter(check))
      {
         switch (check)
         {
            case '_':
            case '@':
            case '#':
               break;

            default:
               return(null);
         }
      }

      Loop:
      while (++pos < text.length())
      {
         check = text.charAt(pos);

         if (Character.isLetter(check))
            continue;

         switch (check)
         {
            case '_':
            case '@':
            case '#':
            case '$':
               break;

            default:
               break Loop;
         }
      }

      position.set(pos);

      result = text.substring(start, pos);

      return(result);
   }  
```

Code after the change...

```
   public static @Nullable String extractIdentifier(String text, MutableInteger position)
   {
      String result;
      int start, pos;

      start = position.get();
      pos   = start;

      if (!isIdentifier(text, pos, true))
         return(null);

      while (++pos < text.length())
      {
         if (!isIdentifier(text, pos, false))
            break;
      }

      position.set(pos);

      result = text.substring(start, pos);

      return(result);
   }
```

**Running PMD through:** Eclipse

**Comment by oowekyala** [comment]
Thanks for the report! But this is a duplicate of #1984, which is already fixed on master

**Comment by numeralnathan** [comment]
@oowekyala I tried looking for duplicates.  Thanks for pointing this out!
